rfcs/067-prismic-api-ids/README.md

```diff
@@ -0,0 +1,46 @@
+
+SliceMachine makes these snake_case by default but allows this to be overridden easily in the Graphical User Interface (GUI) and to-date we have made *all* of them kebab-case, so there probably isn't a good case for changing these to snake_case now.
+
+A couple of reusable content types have been given singular API IDs – `card` and `collection-venue` – should we consider changing these two to `cards` and `collection-venues`?
+
+## Field API IDs
+Field API IDs are individual properties on a Custom type or Slice. More often than not these are individual words (e.g. `title`), which keeps things simple. When there's more than one word, we now have a mixture of snake_case and camelCase field API IDs.
+
+- [Example of a camelCase field](https://github.com/wellcomecollection/wellcomecollection.org/blob/main/common/views/slices/CollectionVenue/model.json#L37)
+- [Example of a snake_case field](https://github.com/wellcomecollection/wellcomecollection.org/blob/main/common/views/slices/GuideStop/model.json#L31)
+
+SliceMachine defaults to snake_case (based on the label), although this is readily overridable in the GUI at the point of adding the field.
+
+We exclusively work with camelCased variables in TypeScript and while snake_cased object properties aren’t a problem (e.g. `data.some_property`), we wouldn’t be able to destructure those properties off the `data` object without it being a linting error:
+
+```
+const { some_property } = data; // <-- linting error
+```
+
+Whereas this isn’t a problem:
+
+```
+const { someProperty } = data; // <-- ✨
+```
+
+So perhaps the preference should be for updating the field names at point of creation to be camelCased to remove the need for an extra renaming variable step in the TypeScript.
+
+## Slice IDs
+Prior to our use of SliceMachine, we gave our Slices camelCased IDs. However, SliceMachine requires that Slices have snake_cased IDs (it doesn’t let you override it in the GUI). When we migrated to SliceMachine we updated the Slice IDs in the [code](https://github.com/wellcomecollection/wellcomecollection.org/blob/main/common/views/slices/index.ts) and [type](https://github.com/wellcomecollection/wellcomecollection.org/blob/main/common/prismicio-types.d.ts) files that it generates to be camelCased in order to be able to use our existing code consistently. But each subsequent change to the Slice through SliceMachine will revert the IDs to snake_case and we have to remember to override it back to camelCase in several places across these two files. For newer Slices we have kept the default (snake_case) ids rather than override them to camelCase. 
+
+I don’t think we deal directly with the Slice IDs. Instead, we send all of them (in [a `components` object](https://github.com/wellcomecollection/wellcomecollection.org/blob/main/common/views/slices/index.ts#L5-L29)) along to SliceMachine, so the linting problems mentioned above don’t arise. As such, I propose that we move to using snake_case for all Slice IDs.
```

Comment:
Agreed. Since it gets overwritten every time, we don't know to revert camelcasing anywhere do we?